### PR TITLE
Dilpreet | Test | Fixed thermal cards

### PR DESCRIPTION
**Pull request by ravinderwalkover**
https://app.clickup.com/t/27qrgpt
https://app.clickup.com/t/27qrhpp
https://app.clickup.com/t/27qrmr1

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "search.exclude": {
         "**/dist": true
     },
+    "angular.enable-strict-mode-prompt": false,
```
Comment (ravinderwalkover) [inline .vscode/settings.json]: revert

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html [hunk 2]
```diff
@@ -242,7 +247,10 @@ <h2>Header</h2>
                                 >
                                     <span class="max-character-label">Characters left:&nbsp;</span>
                                     <span class="max-character-value">{{
-                                        15 - customTemplate.sections["header"].data["invoiceDate"].label.length
+                                        customTemplate.templateType === "gst_template_a" ||
+                                        customTemplate.templateType === "gst_template_e"
+                                            ? 15
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html]:
```suggestion
                                            ? 15 - customTemplate.sections["header"].data["invoiceDate"].label.length
```

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html [hunk 3]
```diff
@@ -276,7 +289,10 @@ <h2>Header</h2>
                                 >
                                     <span class="max-character-label">Characters left:&nbsp;</span>
                                     <span class="max-character-value">{{
-                                        15 - customTemplate.sections["header"].data["invoiceNumber"].label.length
+                                        customTemplate.templateType === "gst_template_a" ||
+                                        customTemplate.templateType === "gst_template_e"
+                                            ? 15
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html]:
```suggestion
                                            ? 15 - customTemplate.sections["header"].data["invoiceNumber"].label.length
```

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html [hunk 4]
```diff
@@ -1122,7 +1106,10 @@ <h2>Table</h2>
                                 >
                                     <span class="max-character-label">Characters left:&nbsp;</span
                                     ><span class="max-character-value">{{
-                                        12 - customTemplate.sections["table"].data["item"].label.length
+                                        customTemplate.templateType === "gst_template_a" ||
+                                        customTemplate.templateType === "gst_template_e"
+                                            ? 12
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html]:
```suggestion
                                            ? 12 - customTemplate.sections["table"].data["item"].label.length
```

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html [hunk 5]
```diff
@@ -1295,7 +1277,12 @@ <h2>Table</h2>
                                     type="text"
                                     hasFocus
                                     #rate="hasFocus"
-                                    maxlength="10"
+                                    [maxlength]="
+                                        customTemplate.templateType === 'gst_template_a' ||
+                                        customTemplate.templateType === 'gst_template_e'
+                                            ? 12
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html]:
```suggestion
                                            ? 10
```

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html [hunk 6]
```diff
@@ -1308,16 +1295,15 @@ <h2>Table</h2>
                                 >
                                     <span class="max-character-label">Characters left:&nbsp;</span
                                     ><span class="max-character-value">{{
-                                        10 - customTemplate.sections["table"].data["rate"].label.length
+                                        customTemplate.templateType === "gst_template_a" ||
+                                        customTemplate.templateType === "gst_template_e"
+                                            ? 12
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html]:
```suggestion
                                            ? 10 - customTemplate.sections["table"].data["rate"].label.length
```

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html [hunk 7]
```diff
@@ -1459,15 +1445,18 @@ <h2>Table</h2>
                                 >
                                     <span class="max-character-label">Characters left:&nbsp;</span
                                     ><span class="max-character-value">{{
-                                        6 - customTemplate.sections["table"].data["total"].label.length
+                                        customTemplate.templateType === "gst_template_a" ||
+                                        customTemplate.templateType === "gst_template_e"
+                                            ? 6
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html]:
```suggestion
                                            ? 6 - customTemplate.sections["table"].data["total"].label.length
```

File: apps/web-giddh/src/app/services/thermal.service.ts [hunk 8]
```diff
@@ -14,7 +14,10 @@ export class ThermalService {
     * @memberof ThermalComponent
     */
     public print(defaultTemplate: any, request: any): void {
+<<<<<<< HEAD
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/thermal.service.ts]: remove the conflict code

File: apps/web-giddh/src/app/services/thermal.service.ts [hunk 9]
```diff
@@ -310,14 +328,18 @@ export class ThermalService {
          * This will use for hide/show for footer total
          */
         let netAmountField;
+<<<<<<< HEAD
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/thermal.service.ts]: remove conflict code

File: apps/web-giddh/src/app/services/thermal.service.ts [hunk 10]
```diff
@@ -535,6 +559,11 @@ export class ThermalService {
                 }
             }
         });
+
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/thermal.service.ts]: remove extra spaces

File: apps/web-giddh/src/app/services/thermal.service.ts [hunk 11]
```diff
@@ -660,6 +689,8 @@ export class ThermalService {
                         this.printerFormat.endPrinter +
                         this.printerFormat.fullCut,
                     ];
+                    console.log(txt);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/thermal.service.ts:675]: remove console after testing

File: apps/web-giddh/src/environments/environment.ts [hunk 12]
```diff
@@ -8,7 +8,7 @@ Error.stackTraceLimit = Infinity;
 // require('zone.js/dist/long-stack-trace-zone');
 
 export const environment: Environment = {
-    production: false,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/environments/environment.ts]: revert this

File: apps/web-giddh/src/app/services/thermal.service.ts [hunk 13]
```diff
@@ -231,13 +232,17 @@ export class ThermalService {
...
          */
         let noOfItemsField;
-        let noOfItems;
+        let noOfItems = 0;
+
         if (defaultTemplate?.sections?.table?.data?.totalQuantity?.display) {
             noOfItemsField = defaultTemplate?.sections?.table?.data?.totalQuantity?.label;
-            noOfItems = request?.entries.length;
+            for (let entry of request.entries) {
+                for (let entryss of entry.transactions) {
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/services/thermal.service.ts]:
```suggestion
                for (let transaction of entry.transactions) {
```

File: apps/web-giddh/src/app/services/thermal.service.ts [hunk 14]
```diff
@@ -247,7 +252,7 @@ export class ThermalService {
         let totalAmountField;
         let totalWords;
         if (defaultTemplate?.sections?.footer?.data?.totalDue?.display && defaultTemplate?.sections?.footer?.data?.totalInWords?.display) {
-            totalAmountField = defaultTemplate?.sections?.footer?.data?.totalDue?.label;
+            totalAmountField = 'Invoice Total';
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/services/thermal.service.ts:255]: Please check this, hard coded value is set

File: apps/web-giddh/src/app/services/thermal.service.ts [hunk 15]
```diff
@@ -266,28 +271,36 @@ export class ThermalService {
...
         /**
          * This will use for hide/show for tax amount
          */
         let taxAmountField;
-        let taxTotal;
+        let taxableAmount = 0;
         if (defaultTemplate?.sections?.table?.data?.taxableValue?.display) {
-            taxTotal = parseFloat(request?.taxTotal?.amountForAccount).toFixed(2);
+            for (let entry of request.entries) {
+                for (let entryss of entry.transactions) {
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/services/thermal.service.ts]:
```suggestion
                for (let transaction of entry.transactions) {
```

**Review by the-prateek-mishra — CHANGES_REQUESTED** [review]

**Review by the-prateek-mishra — APPROVED** [review]

**Pull request merged**